File: java/leetcode/src/main/java/com/leetcode/TwoPointers/Assign_Cookies_455.java
package com.leetcode.TwoPointers;

import java.util.Arrays;

public class Assign_Cookies_455 {
    public int findContentChildren(int[] g, int[] s) {
        Arrays.sort(s);
        Arrays.sort(g);
        int j = 0;
        for (int i = 0; i < s.length && j < g.length; i++) {
            if (s[i] >= g[j]) j++;
        }
        return j;
    }
}

File: js/algrithm/huahua130/two-pointer/Assign_Cookies_455.js
/**
 * @param {number[]} g
 * @param {number[]} s
 * @return {number}
 */
var findContentChildren = function(g, s) {
    g.sort((a, b) => a - b)
    s.sort((a, b) => a - b)

    let i = 0, j = 0
    let cnt = 0
    while (i < g.length && j < s.length) {
        if (g[i] <= s[j]) {
            i++
            j++
            cnt++
        } else {
            j++
        }
    }
    return cnt
};